tests/epyccel/modules/arrays.py

```diff
@@ -1753,3 +1745,21 @@ def multiple_np_linspace():
     y = np.linspace(0, 4, 128)
     z = np.linspace(0, 8, 128)
     return x[0] + y[1] + z[2] + linspace_index
+
+#==============================================================================
+# Iteration
+#==============================================================================
+
+def iterate_slice(i : int):
+    from numpy import arange
+    a = arange(15)
+    res = 0
+    for ai in a[:i]:
+        res += ai
+    return res
+
+def unpack_array():
+    from numpy import zeros
+    arr = zeros(3, dtype='float64')
+    x, y, z = arr[:]
+    return x, y, z
```

Comment:
Can we also get 1D arrays by unpacking a 2D array? We might want to check this too.

Reply:
I have added a 2D test : 671062d83

This is currently limited currently to the case where we know that `arr.shape[0] == 3`. Would you like me to be more generous and allow using this syntax with an array of unknown shape?

Reply:
I think we should support it for 2 reasons:
1. We usually let the user know more about the shape than we do (e.g. in `x+y` we assume that the user knows that `x` and `y` have compatible shapes
2. Our usual rule is that if it runs in Python then it can be translated. If the shape is wrong then this will not run in Python so it is not our fault if the result is unexpected.

I have fixed this : [2d5201d](https://github.com/pyccel/pyccel/pull/1793/commits/2d5201d4246cc72f93f896840cb800ef9dc830ec) and [304747c](https://github.com/pyccel/pyccel/pull/1793/commits/304747c5bccff0aa7eb75000f8b14b4d3b0c5826)

I did wonder if we should raise a warning though? "Assuming shape is (3,...)"
But there are lots of places where we assume shapes and do not raise such warnings

Reply:
I agree with your argument. And indeed we cannot raise warnings everywhere. Thanks for fixing that too!